bpf,init.sh: make netdev bpf filter cleanup less eager

ti-mo opened the pull request:
Due to an oversight when updating init.sh to deal with the new tc filter names for bpf progs after the introduction of Go-based loader/netlink attach, all interfaces in the host namespace that didn't contain the word 'cilium' would have their egress and ingress filters stripped. This included lxc interfaces and many others. lxc programs in particular would only be reattached when the endpoint got regenerated, which can take a while on nodes with many Pods. This caused connectivity interruptions in the meantime.

This commit changes the tc filter naming convention to converge on the changes introduced in 2e40d67ec9 ("bpf: Finish rename of BPF programs to cil_ prefix"), using the bpf program (function) name containing the 'cil_' prefix. The 'cilium_' prefix is no longer included explicitly, instead opting for the program name suffixed by the interface name, e.g. cil_from_netdev-eth0.

init.sh no longer uses the term 'cilium' to trigger a removal of the interface's tc filters.

Fixes commit 2a7cef4bb3 ("init,cleanup: remove TC filters containing 'cilium' in their names").

```release-note
Fix Pod connectivity interruption during agent restart
```

Fixes: https://github.com/cilium/cilium/issues/24191

ti-mo committed:
bpf,init.sh: make netdev bpf filter cleanup less eager

Due to an oversight when updating init.sh to deal with the new tc filter names
for bpf progs after the introduction of Go-based loader/netlink attach, all
interfaces in the host namespace that didn't contain the word 'cilium' would
have their egress and ingress filters stripped. This included lxc interfaces
and many others. lxc programs in particular would only be reattached when the
endpoint got regenerated, which can take a while on nodes with many Pods.
This caused connectivity interruptions in the meantime.

This commit changes the tc filter naming convention to converge on the changes
introduced in 2e40d67ec9 ("bpf: Finish rename of BPF programs to cil_ prefix"),
using the bpf program (function) name containing the 'cil_' prefix. The
'cilium_' prefix is no longer included explicitly, instead opting for the
program name suffixed by the interface name, e.g. cil_from_netdev-eth0.

init.sh no longer uses the term 'cilium' to trigger a removal of the
interface's tc filters. Also switched over to a regex that acts on a word
boundary to reduce the chance of a false positive (e.g. a filter pencil_foo
installed by another tool should not trigger removal).

Fixes commit 2a7cef4bb3 ("init,cleanup: remove TC filters containing 'cilium'
in their names").

Signed-off-by: Timo Beckers <timo@isovalent.com>

brb reviewed (commented):
While we are here, let's extend the K8sUpdates test to check whether flows got interrupted. In particular, we can reuse https://github.com/cilium/cilium/blob/master/test/k8s/updates.go#L328, and call the function after restarting Cilium.

ti-mo committed:
test/k8s: restart agent and check for svc interrupts before downgrade

We would only test service interruptions using up/downgrades, but agent
restarts should also be clean.

Signed-off-by: Timo Beckers <timo@isovalent.com>
Co-authored-by: Martynas Pumputis <m@lambda.lt>
